### DISCUSSION TITLE
Receive a canvas instead of a container

### DISCUSSION THREAD

**Discussion by TiagoCavalcante** [Mathics3/mathics-threejs-backend#73] [Ideas]
Currently `drawGraphics3d` receives up to 4 parameters, where the last 2 are `maxSize` and `innerWidthMultiplier`.

This is very odd, I think this shouldn't be done this way because this way we don't allow things like width != height, parent-element percentage width, CSS media queries, ... and make the window resize slower even when the canvas isn't resized.

`maxSize` and `innerWidthMultiplier` would be replaced by a bit of CSS.

Current version:
```html
<div id="container" />

<script>
    drawGraphics3d(document.getElementById("container", {...}, 400, 0.65)
</script>
```

Proposal:
```html
<style>
    canvas {
        width: 400px;
        height: 400px;
        max-width: 65vw;
        max-height: 65vh;
    }
</style>

<canvas id="graphics" />

<script>
    drawGraphics3d(document.getElementById("graphics", {...})
</script>
```

I could add a "conversion" in the translation layer to keep the backward compatibility.

@rocky @axkr your thoughts?

**Comment by rocky** [comment]
Use css for the win. 

The mechanics would be to bump the threejs version to 1.1.0 if compatibility and 2.0.0 if no compatibility. It looks like we don't separate an API version with the version of the package name itself.

**Comment by axkr** [comment]
> I could add a "conversion" in the translation layer to keep the backward compatibility.
If you provide a translation layer (for a short period of time) this should work.

**Reply by TiagoCavalcante** [reply]
> If you provide a translation layer (for a short period of time) this should work.

Already there is a translation layer for Mathics compatibility. About the short period of time, the "short" needs to be until Sep 11, 2026. I [promised LTS](https://github.com/Mathics3/mathics-threejs-backend/releases/tag/1.0.0), and this would be only a tiny thing in the translation layer.

> The mechanics would be to bump the threejs version to 1.1.0 if compatibility and 2.0.0 if no compatibility.

I'm going to bump it to 1.2.0, 1.1.0 already was released, but before 1.2.0 there are some (a lot?) of things I want to improve.

> It looks like we don't separate an API version with the version of the package name itself.

We do, currently, the API revision is 1.2, and the mathics-threejs-backend version is 1.1.0.

**Comment by TiagoCavalcante** [comment]
Thinking well, it should receive the container as we need to add the tick labels to the container, but the CSS configuration should be in the container, not in mathics-threejs-backend.

**Comment by TiagoCavalcante** [comment]
Implemented in PR #101.
